modules/code-editor/src/views/full/components/UploadModal.tsx

```diff
@@ -0,0 +1,189 @@
+      hasError: false,
+      alreadyExists: false,
+      overwrite: false
+    }
+  } else if (action.type === 'receivedFile') {
+    const { file, filePath, fullPath } = action.data
+    return { ...state, file, filePath, fullPath }
+  } else if (action.type === 'updateLocation') {
+    const { fullPath, alreadyExists } = action.data
+    return { ...state, fullPath, alreadyExists }
+  } else if (action.type === 'changeOverwrite') {
+    const { overwrite } = action.data
+    return { ...state, overwrite }
+  } else if (action.type === 'startUpload') {
+    return { ...state, isLoading: true }
+  } else if (action.type === 'showError') {
+    return { ...state, isLoading: false, hasError: true }
+  } else {
+    throw new Error(`That action type isn't supported.`)
+  }
+}
+
+export const UploadModal: FC<Props> = props => {
+  const [state, dispatch] = React.useReducer(fetchReducer, {
+    file: undefined,
+    fullPath: '',
+    filePath: '',
+    isLoading: false,
+    hasError: false,
+    alreadyExists: false,
+    overwrite: false
+  })
+
+  const { file, fullPath, filePath, isLoading, hasError, alreadyExists, overwrite } = state
+
+  const submitChanges = async () => {
+    dispatch({ type: 'startUpload' })
+
+    try {
+      const form = new FormData()
+      form.append('location', fullPath)
+      form.append('file', file)
+
+      await props.uploadFile(form)
+      closeDialog()
+    } catch (err) {
+      dispatch({ type: 'hasError' })
+      toastFailure(err.message)
+    }
+  }
+
+  const closeDialog = () => {
+    dispatch({ type: 'clearState' })
+    props.toggle()
+  }
+
+  const readFile = (files: FileList | null) => {
+    if (!files) {
+      return
+    }
+
+    dispatch({
+      type: 'receivedFile',
+      data: {
+        file: files[0],
+        filePath: files[0].name,
+        fullPath: state.fullPath.length ? state.fullPath : files[0].name
+      }
+    })
+  }
+
+  const toggleConfirmation = event => {
+    dispatch({ type: 'changeOverwrite', data: { overwrite: event.currentTarget.checked } })
+  }
+
+  const updateLocation = text => {
+    const sanitized = sanitizeName(text)
+    const existing = !!props.files.raw.find(x => x.location === sanitized)
+
+    dispatch({
+      type: 'updateLocation',
+      data: {
+        fullPath: sanitized,
+        alreadyExists: existing
+      }
+    })
+  }
+
+  return (
+    <Dialog
+      title={lang.tr('module.code-editor.uploadModal.uploadFile')}
+      icon="upload"
+      isOpen={props.isOpen}
+      onClose={closeDialog}
+      transitionDuration={0}
+      canOutsideClickClose={false}
+    >
+      <div
+        onDragOver={e => e.preventDefault()}
+        onDrop={e => {
+          e.preventDefault()
+          readFile(e.dataTransfer.files)
+        }}
+      >
+        <div className={Classes.DIALOG_BODY}>
+          <FormGroup
+            label={<span>{lang.tr('module.code-editor.uploadModal.selectFile')}</span>}
+            labelFor="input-archive"
```

Comment:
labelFor usually goes in pair with an named input, I don't see corresponding input